src/features/DebugSession.ts

```diff
@@ -105,7 +114,7 @@ export class PickPSHostProcessFeature implements IFeature {
 
         if (this.waitingForClientToken) {
             this.clearWaitingToken();
-            return this.pickPSHostProcess();
```

Comment:
Yep you shouldn't need to worry about thread synchronization in JavaScript.  If this is good enough for now I'll merge it and then poke around with it a bit.